python/ganga/__init__.py

```diff
@@ -0,0 +1,301 @@
+# Bootstrap all of ganga, setup GPI, registries, etc.
+from Ganga.Utility.Runtime import allRuntimes
+from Ganga.Utility.Config import getConfig
+from Ganga.Utility.logging import getLogger
+from Ganga.Utility.Plugin import allPlugins
+from Ganga.GPIDev.Base import ProtectedAttributeError, ReadOnlyObjectError, GangaAttributeError
+from Ganga.GPIDev.Lib.Job.Job import JobError
+from Ganga import _gangaPythonPath
+from Ganga.GPIDev.Base.Proxy import GPIProxyObjectFactory
+from Ganga.GPIDev.Credentials import getCredential
+from Ganga.GPIDev.Persistency import export, load
+from Ganga.GPIDev.Adapters.IPostProcessor import MultiPostProcessor
+from Ganga.Runtime import Repository_runtime
+import Ganga.Core
+from Ganga.GPIDev.Lib.JobTree import TreeError
+from Ganga.Runtime import Workspace_runtime
+from Ganga.Core.GangaRepository import getRegistry
+from Ganga.GPIDev.Base.VPrinter import full_print
+from Ganga.GPIDev.Base.Proxy import proxyRef
+import Ganga.GPIDev.Lib.Config
+from Ganga.Utility.feedback_report import report
+from Ganga.Runtime.gangadoc import adddoc
+from Ganga.Core.GangaThread.WorkerThreads.ThreadPoolQueueMonitor import ThreadPoolQueueMonitor
+from Ganga.Runtime import plugins
+
+logger = getLogger(modulename=True)
+
+
+def exportToPublicInterface(name, object, doc_section, docstring=None):
+    """
+    export the given functions/objects to both Ganga.GPI and ganga
+    Note that GPI should be retired in time
+    """
+    import ganga
+    import Ganga.GPI
+
+    # export to GPI
+    setattr(Ganga.GPI, name, object)
+    setattr(ganga, name, object)
+    adddoc(name, object, doc_section, docstring)
+
+
+def ganga_license():
+    'Print the full license (GPL)'
+    with open(os.path.join(_gangaPythonPath, '..', 'LICENSE_GPL')) as printable:
+        logger.info(printable.read())
+
+
+def typename(obj):
+    """Return a name of Ganga object as a string, example: typename(j.application) -> 'DaVinci'"""
+    from Ganga.GPIDev.Base.Proxy import isProxy, stripProxy, proxyRef
+    if isProxy(obj):
+        if hasattr(stripProxy(obj), '_name'):
+            return stripProxy(obj)._name
+        else:
+            logger = Ganga.Utility.logging.getLogger()
+            logger.error("Object %s DOES NOT have the _name parameter set" % (str(obj)))
+            return ""
+    else:
+        if hasattr(obj, '_name'):
+            return obj._name
+        else:
+            logger = Ganga.Utility.logging.getLogger()
+            logger.error("Object %s DOES NOT have the %s or _name parameter set" % (str(obj), str(proxyRef)))
+            return ""
```

Comment:
`return stripProxy(obj)._name` should be sufficient for any proxies or `GangaObject`s. Likewise for the `_category` below.